### PR TITLE
#323 Feature/month view

### PR DESCRIPTION
**Pull request by EugeneValko**
Screenshot_1

### CODE REVIEW

File: frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<ng-template #customCellTemplate let-day="day">
+    <div class="menu-trigger" [matMenuTriggerFor]="menu">
+        <div class="cal-cell-top">
+            <span class="cal-day-number">{{ day.date | date :"d" }}</span>
+            <i class="fa-solid fa-repeat" *ngIf="getWeekScheduleTime(day.date).length > 0"></i>
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.html]:
```suggestion
            <i class="fa-solid fa-repeat" *ngIf="getWeekScheduleTime(day.date).length"></i>
```

File: frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        </div>
+        <div>
+            <div class="cal-content">
+                <p *ngIf="getWeekScheduleTime(day.date).length === 0">Unavailable</p>
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.html]:
```suggestion
                <p *ngIf="!getWeekScheduleTime(day.date).length">Unavailable</p>
```

File: frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        this.dialog.open(TimePickerDialogComponent, {
+            data: this.items.filter(item => item.weekDay.toString() === this.getWeekDay(new Date(date)))[0],
+        });
+    }
+
+    getWeekScheduleTime(date: string): string[] {
+        const weekDay = this.getWeekDay(new Date(date));
+
+        return this.items.filter(item => item.weekDay === weekDay && item.isEnabled)
+            .map(item => `${item.start.substring(0, 5)}-${item.end.substring(0, 5)}`);
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.ts]: Could you use date pipe?

File: frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.html (hunk 1 above)
Comment (Pavliuk13) [inline frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.html]:
```suggestion
            <i class="fa-solid fa-repeat" *ngIf="getWeekScheduleTime(day.date).length"></i>
```

File: frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.html (hunk 2 above)
Comment (Pavliuk13) [inline frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.html]:
```suggestion
                <p *ngIf="!getWeekScheduleTime(day.date).length">Unavailable</p>
```

**Review by Pavliuk13 — CHANGES_REQUESTED** [review]

**Review by Pavliuk13 — APPROVED** [review]

File: frontend/src/app/shared/components/time-picker-dialog/time-picker-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    end: string;
+
+    isApplyActive: boolean;
+
+    constructor(
+        public dialogRef: MatDialogRef<TimePickerDialogComponent>,
+        @Inject(MAT_DIALOG_DATA) public scheduleItem: IScheduleItem,
+    ) {
+        this.start = scheduleItem.start.substring(0, 5);
+        this.end = scheduleItem.end.substring(0, 5);
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/time-picker-dialog/time-picker-dialog.component.ts]: see the previous comment

File: frontend/src/app/shared/components/time-picker-dialog/time-picker-dialog.component.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    onClickCancel() {
+        this.dialogRef.close();
+    }
+
+    onClickApply() {
+        this.scheduleItem.start = this.start;
+        this.scheduleItem.end = this.end;
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/time-picker-dialog/time-picker-dialog.component.ts]: Please get rid of mutations

File: frontend/src/app/shared/components/time-picker-dialog/time-picker-dialog.component.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    timeToDate(time: string): Date {
+        const chunks = time.split(':');
+        const date = new Date();
+
+        date.setHours(Number(chunks[0]));
+        date.setMinutes(Number(chunks[1]));
+
+        return date;
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/time-picker-dialog/time-picker-dialog.component.ts]: why do we need this?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.html (hunk 1 above)
Comment (EvgenValko) [reply frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.html]: fix

File: frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.html (hunk 2 above)
Comment (EvgenValko) [reply frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.html]: fix

**Review by tatianahutii — APPROVED** [review]

File: frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.sass [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        .cal-cell
+            color: $dark-gray
+            padding: 25px 0 10px 0
+
+.menu-trigger:hover
+    height: 100%
+    width: 100%
+    cursor: pointer
+
+::ng-deep .mat-menu-panel
```
Comment (apofis96) [inline frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.sass]: add wrapper

**Review by apofis96 — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.sass [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        .cal-cell
+            color: $dark-gray
+            padding: 25px 0 10px 0
+
+.menu-trigger:hover
+    height: 100%
+    width: 100%
+    cursor: pointer
+
+::ng-deep schedule-month-drop-down-menu
```
Comment (apofis96) [inline frontend/src/app/modules/availability/new-slot/schedule-month/schedule-month.component.sass]: this is not class

**Review by apofis96 — CHANGES_REQUESTED** [review]

**Review by apofis96 — APPROVED** [review]

**Pull request merged**